File: sourceC# - Visual Studio/Workspaces/ESHOP/EShop/FrontEnd/src/app/Module/MProductAttribute/ProductAttribute.Entity.ts
import { IEntity } from "../IEntity.Entity";
import { CategoryEntity } from "../MCategory/Category.Entity";
//import { ProductAttributeNameEntity } from "../MProductAttributeName/ProductAttributeName.Entity";
import { ProductValueEntity } from "../MProductValue/ProductValue.Entity";


export class ProductAttributeEntity extends IEntity {
  CategoryId: number = null;
  Code: string = null;
  Id: number = null;
  Cx: number = null;
  CategoryEntity: CategoryEntity = null;
  //ProductAttributeNameEntities: Array<ProductAttributeNameEntity> = [];
  ProductValueEntities: Array<ProductValueEntity> = [];
    public constructor() {
        super();
    }

}

File: sourceC# - Visual Studio/Workspaces/ESHOP/EShop/Entities/ProductAttributeEntity.cs
using EShop.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace EShop.Entities
{
    public class ProductAttributeEntity : BaseEntity
    {
        public Guid Id { get; set; }
        public String Code { get; set; }
        public Guid CategoryId { get; set; }
        public CategoryEntity CategoryEntity { get; set; }
        public List<ProductAttributeNameEntity> ProductAttributeNameEntities { get; set; }
        public List<ProductValueEntity> ProductValueEntities { get; set; }

        public ProductAttributeEntity():base() { }

        public ProductAttributeEntity(ProductAttribute ProductAttribute, params object[] args) :base(ProductAttribute)
        {
		    foreach(object arg in args)
			{
                if (arg is Category Category)
                    CategoryEntity = new CategoryEntity(Category);				
                if (arg is ICollection<ProductAttributeName> ProductAttributeNames)
                    ProductAttributeNameEntities = ProductAttributeNames.Select(model => new ProductAttributeNameEntity(model, model.Language)).ToList();				
                if (arg is ICollection<ProductValue> ProductValues)
                    ProductValueEntities = ProductValues.Select(model => new ProductValueEntity(model, model.Language, model.Product)).ToList();				
			}
        }
    }

    public class ProductAttributeSearchEntity : FilterEntity
    {
        public Guid? Id { get; set; }
        public String Code { get; set; }
        public Guid? CategoryId { get; set; }
    }
}
